### ISSUE TITLE
Fix spandoc in the BlockVerifier tests

### ISSUE THREAD

**Issue by teor2345** [ZcashFoundation/zebra#609]
Blocked by the code movement in ticket #582.

We need to fix spandoc usage in the BlockVerifier tests, so we call `map_err` before `await`:
```
                .await
                .map_err(|e| eyre!(e))?;
```
Becomes:
```
                .map_err(|e| eyre!(e))
                .await?;
```

See commit fec6d2ba47be0fffa9e8c11687a7b94a83136000 for more examples of this fix in CheckpointVerifier.

> So I'm working on fixing this but for now spandoc is restricted on how thoroughly it can instrument futures. The problem is it can only instrument the futures themselves, and not the rest of the expressions in the chain of function calls. The way you have this setup (and the way I setup the examples that you based this on) is subtly wrong. Right here you're constructing a `Report` in a combinator on the `Result` returned from the future, so this gets called after the future is done executing and the span is no longer entered. We need to start using `TryFutureExt::map_err` before the await to have the conversion run as part of the future if we want to get these spandoc comments in our error messages.
> 
> So if you notice your spandoc spans aren't showing up in your eyre::Reports that's why.

_Originally posted by @yaahc in https://github.com/_render_node/MDE3OlB1bGxSZXF1ZXN0UmV2aWV3NDQ0MTM3MDQx/pull_request_reviews/more_threads_

**Comment by teor2345** [comment]
@oxarbitrage you might be interested in doing this fix, but I need to do #582 first.

**Comment by yaahc** [comment]
@oxarbitrage don't worry too much about this issue IMO, I'm trying to fix this more fundamentally by making it so spandoc splits expressions up and instruments them individually so it will mix in `span.enter()` and `fut.instrument(span).await` as necessary to instrument an entire expression regardless of how many awaits there are.

**Comment by oxarbitrage** [comment]
ok, discarded.

**Comment by dconnolly** [comment]
Fixed in #618
